index.ts

```diff
@@ -5,41 +5,62 @@ import fetchRetry from 'fetch-retry';
 import isString from 'lodash.isstring';
 import md5 from 'md5';
 import ms from 'ms';
-import fetch from 'node-fetch';
+import fetch, { Response } from 'node-fetch';
 import removeTrailingSlash from 'remove-trailing-slash';
 import { v4 as uuid } from 'uuid';
 
 const version = require('./package.json').version;
 
 const retryableFetch = fetchRetry(fetch as any) as unknown as typeof fetch;
 const setImmediate = global.setImmediate || process.nextTick.bind(process);
-const noop = () => {};
 
-enum AnalyticsState {
+export type AnalyticsMessage = AnalyticsIdentity & {
+  context?: { [key: string]: unknown };
+  traits?: { [key: string]: unknown };
+  integrations?: { [destination: string]: boolean };
+  timestamp?: Date;
+  [key: string]: unknown;
+};
+
+export type AnalyticsIdentity = { userId: string } | { anonymousId: string };
+
+export type AnalyticsMessageCallback = (error?: Error) => void;
```

Comment:
Great catch here - I really thought flush and non-flush methods shared the same callback signature until i dug into the source code to check this typing out.

Reply:
circling back to this just to make it explicit- segment library's functionality for the callbacks is different from rudderstack's. 

In segment:
all events accept a callback of type `AnalyticsFlushCallback`. During `enqueue`, if the method was invoked with a callback, it's [passed](https://github.com/segmentio/analytics-node/blob/master/index.js#L214) into `flush` as the callback for the flush. Then, when `flush` is mapping through all the callbacks, it ignores the second parameter (data) and invokes each callback only with an `err` argument.

In other words, you can call `track((err, data) => cleanupFlush(err, data))` and expect that flush will be called with that callback. As far as I can tell, this callback would be called twice in this case, once for the track, and once for the flush.

In rudderstack:
all events accept a callback of type `AnalyticsMessageCallback `.  The callback is not passed into the flush method during the enqueue method. Flush will only execute a callback if the method was called directly a callback argument e.g. `flush((err, data) => cleanupFlush(err, data))`. 

You cannot expect `track((err, data) => cleanupFlush(err, data))` to call flush with that callback.